### PR TITLE
style: remove redundant instance arguments

### PR DESCRIPTION
**Pull request by JovanGerb**
I removed some redundant instance arguments throughout Mathlib. To do this, I used VS Code's regex search. See https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/repeating.20instances.20from.20variable.20command

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Basic.lean [hunk 1]
```diff
@@ -68,8 +68,7 @@ theorem norm_zsmul [NormedSpace 𝕜 E] (n : ℤ) (x : E) : ‖n • x‖ = ‖(
   rw [← norm_smul, ← Int.smul_one_eq_coe, smul_assoc, one_smul]
 #align norm_zsmul norm_zsmul
 
-variable [SeminormedAddCommGroup E] [NormedSpace 𝕜 E]
-variable [SeminormedAddCommGroup F] [NormedSpace 𝕜 F]
+variable [NormedSpace 𝕜 E] [NormedSpace 𝕜 F]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/NormedSpace/Basic.lean:71]: I would move these up to the line `variable [NormedField 𝕜] [SeminormedAddCommGroup E] [SeminormedAddCommGroup F]`

File: Mathlib/Data/Matrix/Basic.lean [hunk 2]
```diff
@@ -1290,11 +1290,11 @@ theorem scalar_inj [Nonempty n] {r s : α} : scalar n r = scalar n s ↔ r = s :
...
 #align matrix.scalar_inj Matrix.scalar_inj
 
-theorem scalar_commute_iff [Fintype n] [DecidableEq n] {r : α} {M : Matrix n n α} :
+theorem scalar_commute_iff {r : α} {M : Matrix n n α} :
     Commute (scalar n r) M ↔ r • M = MulOpposite.op r • M := by
   simp_rw [Commute, SemiconjBy, scalar_apply, ← smul_eq_diagonal_mul, ← op_smul_eq_mul_diagonal]
 
-theorem scalar_commute [Fintype n] [DecidableEq n] (r : α) (hr : ∀ r', Commute r r')
+theorem scalar_commute (r : α) (hr : ∀ r', Commute r r')
     (M : Matrix n n α) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Matrix/Basic.lean:1298]:
```suggestion
theorem scalar_commute (r : α) (hr : ∀ r', Commute r r') (M : Matrix n n α) :
```

File: Mathlib/GroupTheory/Exponent.lean [hunk 3]
```diff
@@ -447,7 +447,7 @@ theorem exponent_ne_zero_of_finite [Finite G] : exponent G ≠ 0 :=
 #align add_monoid.exponent_ne_zero_of_finite AddMonoid.exponent_ne_zero_of_finite
 
 @[to_additive AddMonoid.one_lt_exponent]
-lemma one_lt_exponent [LeftCancelMonoid G] [Finite G] [Nontrivial G] :
+lemma one_lt_exponent [Finite G] [Nontrivial G] :
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/Exponent.lean:450]: Looks like `[Finite G]` can move up too

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean [hunk 4]
```diff
@@ -1145,7 +1145,7 @@ lemma MeasurableEmbedding.borelSpace {α β : Type*} [MeasurableSpace α] [Topol
   have : MeasurableSpace.comap e (borel β) = ‹_› := by simpa [hβ.measurable_eq] using h'e.comap_eq
   rw [← this, ← borel_comap, h''e.induced]
 
-instance _root_.ULift.instBorelSpace [BorelSpace α] : BorelSpace (ULift α) :=
+instance _root_.ULift.instBorelSpac : BorelSpace (ULift α) :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean:1148]:
```suggestion
instance _root_.ULift.instBorelSpace : BorelSpace (ULift α) :=
```

File: Mathlib/RingTheory/Ideal/QuotientOperations.lean [hunk 5]
```diff
@@ -58,7 +58,7 @@ theorem lift_injective_of_ker_le_ideal (I : Ideal R) {f : R →+* S} (H : ∀ a
 #align ring_hom.lift_injective_of_ker_le_ideal RingHom.lift_injective_of_ker_le_ideal
 
 /-- The induced map from the quotient by the kernel is injective. -/
-theorem kerLift_injective [Semiring S] (f : R →+* S) : Function.Injective (kerLift f) :=
+theorem kerLift_injective (f : R →+* S) : Function.Injective (kerLift f) :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Ideal/QuotientOperations.lean:61]:
```suggestion
theorem kerLift_injective : Function.Injective (kerLift f) :=
```

File: Mathlib/RingTheory/IntegralRestrict.lean [hunk 6]
```diff
@@ -396,8 +396,7 @@ variable [IsDomain Aₘ] [IsIntegrallyClosed Aₘ] [IsDomain Bₘ] [IsIntegrally
 variable [NoZeroSMulDivisors Aₘ Bₘ] [Module.Finite Aₘ Bₘ]
 variable [IsSeparable (FractionRing Aₘ) (FractionRing Bₘ)]
 
-lemma Algebra.intNorm_eq_of_isLocalization
-    [IsSeparable (FractionRing Aₘ) (FractionRing Bₘ)] (x : B) :
+lemma Algebra.intNorm_eq_of_isLocalization  (x : B) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/IntegralRestrict.lean:399]:
```suggestion
lemma Algebra.intNorm_eq_of_isLocalization (x : B) :
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks!

**Comment by Ruben-VandeVelde** [comment]
But you should close this PR and reopen from a branch in the main repository

**Comment by JovanGerb** [comment]
I think I need permission to make a branch in the main repository

**Pull request closed without merging**